src/prefect/settings.py

```diff
@@ -446,6 +449,129 @@ def settings_customise_sources(
             ProfileSettingsTomlLoader(settings_cls),
         )
 
+    @classmethod
+    def valid_setting_names(cls) -> Set[str]:
+        """
+        A set of valid setting names, e.g. "PREFECT_API_URL" or "PREFECT_API_KEY".
+        """
+        settings_fields = set()
+        for field_name, field in cls.model_fields.items():
+            if inspect.isclass(field.annotation) and issubclass(
+                field.annotation, PrefectBaseSettings
+            ):
+                settings_fields.update(field.annotation.valid_setting_names())
+            else:
+                settings_fields.add(
+                    f"{cls.model_config.get('env_prefix')}{field_name.upper()}"
+                )
+        return settings_fields
+
+    def to_environment_variables(
+        self,
+        exclude_unset: bool = False,
+        include_secrets: bool = True,
+    ) -> Dict[str, str]:
+        """Convert the settings object to a dictionary of environment variables."""
+
+        env: Dict[str, Any] = self.model_dump(
+            exclude_unset=exclude_unset,
+            mode="json",
+            context={"include_secrets": include_secrets},
+        )
+        env_variables = {}
+        for key in self.model_fields.keys():
+            if isinstance(child_settings := getattr(self, key), PrefectBaseSettings):
+                child_env = child_settings.to_environment_variables(
+                    exclude_unset=exclude_unset,
+                    include_secrets=include_secrets,
+                )
+                env_variables.update(child_env)
+            elif (value := env.get(key)) is not None:
+                env_variables[
+                    f"{self.model_config.get('env_prefix')}{key.upper()}"
+                ] = str(value)
```

Comment:
The fix for `to_environment_variables` was to iterate over the `model_fields` because nested model changes weren't included when `exclude_unset=True`.